Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm

```diff
@@ -1261,8 +1276,13 @@ static bool isCopyDisplayedPixelBufferAvailable()
             didBecomeReadyForMoreSamples(m_enabledVideoTrackID);
     });
     m_decompressionSession->notifyWhenHasAvailableVideoFrame([weakThis = WeakPtr { *this }, this] {
-        if (weakThis)
+        if (weakThis) {
             setHasAvailableVideoFrame(true);
+            if (m_hasAvailableVideoFrameSemaphore) {
+                m_hasAvailableVideoFrameSemaphore->signal();
+                m_hasAvailableVideoFrameSemaphore = nullptr;
+            }
+        }
     });
```

Comment:
This is repeated in `flushVideo`. While it isn't a lot of code, it would be better to share the code if possible.